doc/modules/cassandra/pages/architecture/overview.adoc

```diff
@@ -38,64 +24,57 @@ following design objectives:
-language, to create and update database schema and access data. CQL
-allows users to organize data within a cluster of Cassandra nodes using:
+{cassandra} provides the Cassandra Query Language (xref:cassandra:developing/cql/ddl.adoc[CQL]), an SQL-like
+language, to create, modify, and delete database schema, as well as access data. 
+CQL allows users to organize data within a cluster of {cassandra} nodes using:
 
-* *Keyspace*: Defines how a dataset is replicated, per datacenter.
+* Keyspace: Defines how a dataset is replicated, per datacenter.
 Replication is the number of copies saved per cluster.
 Keyspaces contain tables.
-* *Table*: Defines the typed schema for a collection of partitions.
-Tables contain partitions, which contain rows, which contain columns.
-Cassandra tables can flexibly add new columns to tables with zero downtime.
-* *Partition*: Defines the mandatory part of the primary key all rows in
-Cassandra must have to identify the node in a cluster where the row is stored.
+* Table: Tables are composed of rows and columns.
+Columns define the typed schema for a single datum in a table.
+Tables are partitioned based on the columns provided in the partition key.
+{cassandra} tables can flexibly add new columns to tables with zero downtime.
+* Partition: Defines the mandatory part of the primary key all rows in
+{cassandra} must have to identify the node in a cluster where the row is stored.
 All performant queries supply the partition key in the query.
-* *Row*: Contains a collection of columns identified by a unique primary
+* Row: Contains a collection of columns identified by a unique primary
 key made up of the partition key and optionally additional clustering
 keys.
-* *Column*: A single datum with a type which belongs to a row.
+* Column: A single datum with a type which belongs to a row.
 
 CQL supports numerous advanced features over a partitioned dataset such
 as:
 
+* Collection types including sets, maps, and lists
+* User-defined types, tuples, functions and aggregates
+* Storage-attached indexing (SAI) for secondary indexes
+* Local secondary indexes (2i)
+* User-defined types, functions and aggregates
 * Single-partition lightweight transactions with atomic compare and set
 semantics
-* User-defined types, functions and aggregates
-* Collection types including sets, maps, and lists
-* Local secondary indices
 * (Experimental) materialized views
 
-Cassandra explicitly chooses not to implement operations that require
+{cassandra} explicitly chooses not to implement operations that require
 cross-partition coordination as they are typically slow and hard to
-provide highly available global semantics. For example Cassandra does
-not support:
+provide highly available global semantics. 
+For example, {cassandra} does not support:
 
 * Cross-partition transactions
 * Distributed joins
 * Foreign keys or referential integrity.
```

Comment:
later on, it might be worth adding a sentence here that this changes in 5.1 with Accord ACID transactions